### ISSUE TITLE
Add diagnostics to preset-env's `useBuiltIns`

### ISSUE THREAD

**Issue by fabiospampinato** [babel/babel#7698]
The `useBuiltIns` option of `preset-env` may load unnecessary polyfills, for instance with the following code:

```js
(function () {
  const obj = {
    name: 'obj'
  };
  function foo ( x ) {
    x.name;
  }
  foo ( obj );
}())
```

it thinks that I need a polyfill for `es6.function.name`, just because I'm accessing a property named "name". Loading the "wrong" polyfill may result in pretty large bundles because of the way core-js works (https://github.com/zloirock/core-js/issues/388#issuecomment-377818011), so we should avoid loading anything that's not necessary.

With small code snippets like the one above it's easy to spot why babel thinks I need those specific polyfills, but with large bundles this is non-trivial.

I would like to have some kind of diagnostics for `useBuiltIns`, basically I would like to know why babel thinks each of those polyfills are required. I think this could help reduce bundle sizes by a significant amount.

**Comment by zloirock** [comment]
Just interesting, which kind of diagnostics? `preset-env` have `debug` option which shows the list of injected polyfills. Do you wanna see the list of code lines which causes injection of this polyfill for each polyfill in the debug output?

**Comment by fabiospampinato** [comment]
> Do you wanna see the list of code lines which causes injection of this polyfill for each polyfill in the debug output?

Yes that's what I was thinking when I opened this issue, but maybe it would not be that useful since a sizable (?) fraction of the code is there to provide maximally strictness and support for ES3 engines as you explained [here](https://github.com/zloirock/core-js/issues/388#issuecomment-377818011).
